platform/windows/display_server_windows.cpp

```diff
@@ -1877,11 +1877,16 @@ LRESULT DisplayServerWindows::WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARA
 			break;
 		}
 		case WM_ACTIVATE: { // Watch For Window Activate Message
-			saved_wparam = wParam;
-			saved_lparam = lParam;
 
-			// Run a timer to prevent event catching warning if the window is closing.
-			focus_timer_id = SetTimer(windows[window_id].hWnd, 2, USER_TIMER_MINIMUM, (TIMERPROC) nullptr);
+			if (!windows[window_id].window_focused) {
+				_process_activate_event(window_id, wParam, lParam);
+			} else {
+				saved_wparam = wParam;
+				saved_lparam = lParam;
+				saved_window_id = window_id;
```

Comment:
Why this is necessary? Time should be associated with the window and have valid `window_id`, or is it redirected to the parent window somehow?

Also, `save_xparam` and `timer_id` probably should be in the window data, instead of global (in case WM_ACTIVATE is received by multiple windows before WM_TIMER is call).

Reply:
Done, I think. I preserve `saved_window_id` but renamed it to `last_activated_window` cuz popups still overlap each other without it. Added params and timers to window data as you said.